lib/colorls/flags.rb

```diff
@@ -6,25 +6,26 @@ class Flags
     def initialize(*args)
       @args = args
       @light_colors = false
+      @mode = :one_per_line unless STDOUT.tty?
+
       @opts = {
         show: false,
         sort: false,
         all: false,
         almost_all: false,
-        report: false,
-        one_per_line: !STDOUT.tty?,
-        long: false,
-        tree: false,
         git_status: false,
         colors: []
       }
 
       parse_options
+
+      # handle mutual exclusive options
+      %i[report tree long one_per_line].each do |value|
```

Comment:
Why are `report` and `long` exclusive?

Reply:
Of course, you're right. I fixed that.

But I am not sure what to do about `--tree` and `--report`. Why is it not a good idea to use both together?

Reply:
If `--tree` and `--report` are used together, any result of report the user sees won't be an intuitive one right? As this is a recursive list, just plainly reporting about the root dir isn't intuitive. And neither is displaying report for entire recursion (basically, after recursion, number of directories = 0).